lib/galaxy/managers/histories.py

```diff
@@ -342,17 +340,13 @@ def get_sharing_extra_information(
         return extra
 
     def is_history_shared_with(self, history: model.History, user: model.User) -> bool:
-        return bool(
-            self.session()
-            .query(self.user_share_model)
-            .filter(
-                and_(
-                    self.user_share_model.table.c.user_id == user.id,
-                    self.user_share_model.table.c.history_id == history.id,
-                )
-            )
-            .first()
+        stmt = (
+            select(HistoryUserShareAssociation.id)
+            .where(HistoryUserShareAssociation.user_id == user.id)
+            .where(HistoryUserShareAssociation.history_id == history.id)
+            .limit(1)
         )
+        return bool(self.session().execute(stmt).first())
```

Comment:
Seems like an exists would be marginally better ?

Reply:
Yes, and easier to read too! I'll add this to the next batch (#16932). 
Thanks for reviewing!

Reply:
Addressed in 7769ddcb8c832b33b88c0c689cf6abee4858a7eb.